app/src/main/java/com/example/androidDeviceDetails/cooker/LocationCooker.kt

```diff
@@ -19,34 +18,31 @@ class LocationCooker : BaseCooker() {
     private val geoCoder: Geocoder = Geocoder(DeviceDetailsApplication.instance)
 
 
-    private fun cookData(locationList: ArrayList<LocationModel>): ArrayList<LocationDisplayModel> {
+    private fun cookData(locationList: ArrayList<LocationModel>): ArrayList<LocationItemViewHolder> {
         val cookedLocationList = emptyList<String>().toMutableList()
         var prevLocationHash = ""
-        for (location in locationList) {
-            val newHash =
-                GeoHash.encodeHash(location.latitude!!, location.longitude!!, geoHashLength).toString()
+        for (loc in locationList) {
+            val newHash = GeoHash.encodeHash(loc.latitude, loc.longitude, geoHashLength).toString()
             if (newHash != prevLocationHash) {
                 prevLocationHash = newHash
                 cookedLocationList.add(newHash)
             }
         }
         val countedData = cookedLocationList.groupingBy { it }.eachCount()
-        val locationDisplayModel: ArrayList<LocationDisplayModel> = ArrayList()
+        val locationItemViewHolder: ArrayList<LocationItemViewHolder> = ArrayList()
```

Comment:
- directly initialise =  ArrayList<LocationItemViewHolder>( )
- this is a list so renave accordingly

Reply:
done